### DISCUSSION TITLE
Not understanding spamhaus error/info

### DISCUSSION THREAD

**Discussion by tve** [docker-mailserver/docker-mailserver#2917] [Q&A]
For a bunch of weeks I've been having inbound mail rejected with a message I'm too stupid to decipher:
```
Nov 30 17:15:19 mail postfix/smtpd[56355]: NOQUEUE: reject: RCPT from p1-182242.info4.citi.com[72.166.182.242]: 554 5.7.1 Service unavailable; Client host [72.166.182.242] blocked using zen.spamhaus.org; Error: open resolver; https://www.spamhaus.org/returnc/pub/44.242.180.137; from=<xxx@info4.citi.com> to=<xxx@VONEICKEN.COM> proto=ESMTP helo=<p1-182242.info4.citi.com>
Nov 30 17:21:26 mail postfix/smtpd[57435]: NOQUEUE: reject: RCPT from out32-17.sg.b.dm.aliyun.com[8.219.32.17]: 554 5.7.1 Service unavailable; Client host [8.219.32.17] blocked using zen.spamhaus.org; Error: open resolver; https://www.spamhaus.org/returnc/pub/34.221.183.230; from=<transaction@notice.aliexpress.com> to=<xxx@voneicken.com> proto=ESMTP helo=<out32-17.sg.b.dm.aliyun.com>
```
Following the link in the error message I get the following info:

> Queries cannot successfully be made to the Spamhaus free infrastructure via public/open resolvers. This is to protect the infrastructure from abuse by large-volume queriers. If you’d like to take a deeper dive into this, check out [successfully accessing Spamhaus’ free blocklists using a public DNS](https://www.spamhaus.com/resource-center/successfully-accessing-spamhauss-free-block-lists-using-a-public-dns/%22).

> Some users continue to query Spamhaus blocklists via public resolvers, unaware that this means that our data does not actually protect their mail stream. We have introduced an error code for these users to provide a clear signal that there is an issue, and that the mailserver configuration needs to be updated.

Is this a bug in docker-mailserver? Am I configuring something wrong? Is this an FAQ that I have not found?

(I'm asking in the discussions in case this has an obvious answer, if it doesn't, I'm happy to provide all the necessary technical info about the set-up. I just restarted the container based on the latest today, so this is an up-to-date version.)

**Comment by tve** [comment]
I've now found https://www.spamhaus.com/resource-center/successfully-accessing-spamhauss-free-block-lists-using-a-public-dns/ and it looks like the errors I'm getting are either due to using a public DNS resolver or due to being hosted in AWS. Not yet sure what the fix is...

I've opened #2918 as I believe DMS needs a fix.

**Comment by tve** [comment]
I believe I finally got to the bottom of this:
- the error message came from postfix, not spamassassin
- I had an ancient override for `smtpd_recipient_restrictions` that listed `reject_rbl_client zen.spamhaus.org` instead of the newer `reject_rbl_client zen.spamhaus.org=127.0.0.[2..11]`, I don't know what the latter does, more sleuthing to do...
- eliminating my override seems to have brought everything back into a working state...

**Comment by tve** [comment]
Looks like I rejoiced too soon, sometimes spamhaus works and sometimes it doesn't. I reopened #2918...

**Comment by georglauterbach** [comment]
See https://github.com/docker-mailserver/docker-mailserver/issues/2918#issuecomment-1335287973
